src/output/workspace-impl.cpp

```diff
@@ -483,7 +483,8 @@ class output_viewport_manager_t
         }
 
         auto old_grid = grid;
-        grid = {vwidth_opt, vheight_opt};
+        grid = {vwidth_opt > 0 ? vwidth_opt : 1,
```

Comment:
This clamping is not necessary, wf-config will sanitize the values once the option has a min/max in the .xml file